src/libpgagroal/pool.c

```diff
@@ -364,9 +369,27 @@ pgagroal_return_connection(int slot, SSL* ssl, bool transaction_mode)
 {
    int state;
    struct configuration* config;
+   time_t now;
+   signed char in_use;
+   signed char age_check;
 
    config = (struct configuration*)shmem;
 
+   /* Kill the connection, if it lives longer than max_connection_age */
+   now = time(NULL);
```

Comment:
We need some extra checks here, right ?

Is `max_connection_age > 0`

Reply:
Thanks for the review! I've tested thses cases this time:
- `idle_timeout` off, `max_connection_age` on, one prefill, no connection / get one connection and then close
- `idle_timeout` on, `max_connection_age` off, one prefill, get one connection and then close
- `idle_timeout = 20`, `max_connection_age = 40`, one prefill, get two connections and then close

Reply:
Excellent